keyext.slicing/src/main/java/org/key_project/slicing/DependencyTracker.java

```diff
@@ -219,7 +221,7 @@ public void ruleApplied(ProofEvent e) {
         for (NodeReplacement newNode : ruleAppInfo.getReplacementNodes()) {
             for (NoPosTacletApp newRule : newNode.getNode().getLocalIntroducedRules()) {
                 if (newRule.rule() instanceof Taclet
-                        && ((Taclet) newRule.rule()).getAddedBy() == n) {
+                        && newNode.getNode() == n) {
```

Comment:
It won't work like that. The purpose of createdBy is to track the node that introduced a new rule. I think this could also be done via some manager registered with the Proof.

Reply:
I think the change above this one looks good.

Reply:
This should work:
```java
                if (newRule.rule() instanceof Taclet taclet) {
                    var justification =
                        n.proof().getInitConfig().getJustifInfo().getJustification(taclet);
                    if (justification instanceof RuleJustificationByAddRules rjbad
                            && rjbad.node() == n) {
                        AddedRule ruleNode = new AddedRule(newRule.rule().name().toString());
                        output.add(ruleNode);
                        dynamicRules.put((Taclet) newRule.rule(), ruleNode);
                    }
                }
```